### PR TITLE
Pre and post configuration scripts

### PR DESCRIPTION
**Pull request by alknopfler**
- launch pre-configuration script
- launch post-configuration script
- Skeleton to add new scripts splitted in functions

### CODE REVIEW

**Comment by alknopfler** [comment]
image
assuming configuration is a file but not executed

File: sztp-agent/pkg/secureAgent/daemon.go [hunk 1]
```diff
@@ -186,3 +195,65 @@ func (a *Agent) downloadAndValidateImage() error {
...
+	if err != nil {
+		log.Println("[ERROR] running the  pre-configuration script", err.Error())
+		return err
+	}
+	log.Println(string(out)) //remove it
+	log.Println("[INFO] Pre-Configuration script executed successfully")
+	return nil
+}
+
+func (a *Agent) launchPostConfiguration() error {
```
Comment (glimchb) [inline sztp-agent/pkg/secureAgent/daemon.go:230]: consider reducing code duplication and create a function here that both `pre` and `post` call

**Review by glimchb — APPROVED** [review]
this is great, approved
consider please reduce code duplication and add a common function here

File: sztp-agent/pkg/secureAgent/daemon.go (hunk 1 above)
Comment (alknopfler) [reply sztp-agent/pkg/secureAgent/daemon.go:230]: [https://github.com/opiproject/sztp/pull/81](https://github.com/opiproject/sztp/pull/81)

**Pull request merged**